caffe-grid/pom.xml

```diff
@@ -18,11 +18,28 @@
     <maven.compiler.target>1.7</maven.compiler.target>
     <hadoop.version>2.5.0</hadoop.version>
     <scala.version>2.10.4</scala.version>
```

Comment:
It seems needed:  The pom.xml has two different types of requiements

here we need scala.version:

    <dependency>
      <groupId>org.scala-lang</groupId>
      <artifactId>scala-library</artifactId>
      <version>${scala.version}</version>
      <scope>provided</scope>

Here we need scala.major.version:

    <dependency>
      <groupId>org.apache.spark</groupId>
      <artifactId>spark-sql_${scala.major.version}</artifactId>
      <version>${spark.version}</version>
      <scope>provided</scope>
    </dependency>